File: src/frontend/src/app/components/cadastro/pessoa/model/pessoa-grid-dto.ts
export class PessoaGridDTO {
  id: string;
  nome: string;
  documento: string; // CPF ou CNPJ formatado
  tipoPessoa: 'FISICA' | 'JURIDICA';
  ativa: boolean;
  createdAt: Date;

  constructor(
    id: string,
    nome: string,
    documento: string,
    tipoPessoa: 'FISICA' | 'JURIDICA',
    ativa: boolean,
    createdAt: Date
  ) {
    this.id = id;
    this.nome = nome;
    this.documento = documento;
    this.tipoPessoa = tipoPessoa;
    this.ativa = ativa;
    this.createdAt = createdAt;
  }
}

File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/cadastro/pessoa/PessoaGridDTO.java
package br.com.grupopipa.gestaointegrada.cadastro.pessoa;

import java.time.LocalDateTime;
import java.util.UUID;

import br.com.grupopipa.gestaointegrada.core.dto.GridDTO;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Builder
@Getter
@Setter
public class PessoaGridDTO implements GridDTO {
    private UUID id;
    private String nome;
    private String documento; // CPF ou CNPJ formatado
    private String tipoPessoa;
    private Boolean ativa;
    private LocalDateTime createdAt;
    private Boolean deleted;
}
